concent_api/middleman/middleman_server.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+import sys
+from logging import getLogger
+
+import signal
+
+from middleman.constants import DEFAULT_INTERNAL_PORT, EXIT_STATUS_ADDRESS_IN_USE_ERROR
+from middleman.constants import LOCALHOST_IP
+from middleman.decorators import safe_execution
+
+logger = getLogger(__name__)
+crash_logger = getLogger('crash')
+
+
+class MiddleMan:
+    def __init__(self, bind_address=None, port=None, loop=None):
+        self._bind_address = bind_address if bind_address is not None else LOCALHOST_IP
+        self._port = port if bind_address is not None else DEFAULT_INTERNAL_PORT
+        self._connection_counter = 0
+        self._server_for_concent = None
+        self._loop = loop if loop is not None else asyncio.get_event_loop()
+
+        # Handle shutdown signal.
+        self._loop.add_signal_handler(signal.SIGTERM, self._terminate_connection)
+
+    def run(self) -> None:
+        try:
+            # start MiddleMan server
+            logger.info("Starting MiddleMan")
+            self._start_middleman()
+        except OSError as exception:
```

Comment:
What is a `KeyboardInterrupt` happens here?

I think there should be a top-level handler that just prints and error and exits when a `KeyboardInterrupt` is detected. We don't want the application to crash and sent a report to Sentry on Ctrl+C.